Sources/JExtractSwift/Swift2JavaGenerator.swift

```diff
@@ -0,0 +1,3 @@
+protocol Swift2JavaGenerator {
+  func generate() throws
```

Comment:
It seems kind of weird that this protocol just has an "empty" function with no arguments. But, I did that to prevent having to pass `AnalysisResult` to a bunch of sub-functions in `print...` So instead the implementations can receive them as initializer arguments.

Reply:
Yeah that’s pretty equivalent. The protocol being just the generate method seemed nicer to me, and we’d swap out what we generate with but it’s equivalent to that if statement. Since there’s not much to share either approach is fine i think, modeling wise i like the protocol but we could do the if